commcare/src/main/java/org/motechproject/commcare/tasks/builder/FormTriggerBuilder.java

```diff
@@ -99,9 +99,13 @@ private TriggerEventRequest buildTriggerForForm(Config config, CommcareApplicati
         String formName = form.getFormName();
 
         String displayName = DisplayNameHelper.buildDisplayName(RECEIVED_FORM, formName, applicationName, config.getName());
-
-        return new TriggerEventRequest(displayName, FORMS_EVENT + "." + config.getName() + "." + form.getXmlns(),
-                null, buildTriggerParameters(form), FORMS_EVENT);
+        if (application.getCommcareAppId() != null) {
+            return new TriggerEventRequest(displayName, FORMS_EVENT + "." + config.getName() + "." + form.getXmlns() + application.getCommcareAppId(),
+                    null, buildTriggerParameters(form), FORMS_EVENT);
+        } else {
+            return new TriggerEventRequest(displayName, FORMS_EVENT + "." + config.getName() + "." + form.getXmlns(),
```

Comment:
first of all,  there should be one return statement. Secondly it should be better to verify if commcareAppId is null in one line